totalRP3_Extended/Misc/Sounds.lua

```diff
@@ -64,88 +84,90 @@ local function isInRadius(maxDistance, posY, posX, myPosY, myPosX)
 	return distance <= maxDistance and distance <= myMaxDistance;
 end
 
-local function isLocalSoundAllowed(sender)
+local function IsLocalSoundAllowed(sender, instanceGUID)
+	local inInstance = IsInInstance();
+
 	-- No instance: sound can be played
-	if not IsInInstance() then
+	if not inInstance then
 		return true;
 	end
 
+	-- Housing instance: check that we're in the same zone
+	if instanceGUID then
+		local currentInstanceGUID = GetHousingInstanceGUID();
+		return currentInstanceGUID == instanceGUID;
+	end
+
 	-- Instance: sound can only be played if in a party or raid
 	local senderAmbiguated = Ambiguate(sender, "none");
-	if UnitInParty(senderAmbiguated) or UnitInRaid(senderAmbiguated) then
+	if UnitIsUnit(sender, "player") or UnitInParty(senderAmbiguated) or UnitInRaid(senderAmbiguated) then
 		return true;
 	end
 
 	return false;
 end
 
+AddOn_TotalRP3.Enums.SOUND_TYPE = {
+	MUSIC = 0,
+	SOUND = 1,
+	SOUND_FILE = 2,
+}
+
+local function PlayLocalSound(soundType, sender, soundID, channel, distance, instanceID, posY, posX, posZ, instanceGUID)
+	if (soundType ~= AddOn_TotalRP3.Enums.SOUND_TYPE.MUSIC and not getConfigValue(TRP3_API.extended.CONFIG_SOUNDS_ACTIVE) or
+		soundType == AddOn_TotalRP3.Enums.SOUND_TYPE.MUSIC and not getConfigValue(TRP3_API.extended.CONFIG_MUSIC_ACTIVE)) then
+		return;
+	end
+
+	if not IsLocalSoundAllowed(sender, instanceGUID) then
+		return;
+	end
+
+	if not (soundID and channel and distance and instanceID and posY and posX and posZ) then
```

Comment:
I'd say sanity checks against data this malformed should be the first check in the function given they'll be cheaper to filter out than a config lookup/three Unit API calls.